test/tests/functional/pbs_sched_preempt_enforce_resumption.py

```diff
@@ -0,0 +1,121 @@
+    """
+    Test sched_preempt_enforce_resumption working
+    """
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # Set default resources available on the default mom
+        a = {ATTR_rescavail + '.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        # Create an express queue
+        b = {ATTR_qtype: 'Execution', ATTR_enable: 'True',
+             ATTR_start: 'True', ATTR_p: '151'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, b, "expressq")
+        a = {ATTR_sched_preempt_enforce_resumption: True}
+        self.server.manager(MGR_CMD_SET, SCHED, a)
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+    def test_filler_job(self):
+        """
+        This test confirms that the filler job does not run if it conflicts
+        with running of a suspended job.
+        """
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 80})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_q: 'expressq',
+                           ATTR_l + '.walltime': 30})
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+
+        j3 = Job(TEST_USER)
+        j3.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_l + '.walltime': 90})
+        jid3 = self.server.submit(j3)
+        starttime = int(time.time())
+        self.scheduler.log_match(
+            "Job would conflict with reservation or top job",
+            starttime=starttime)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid3)
+
+    def test_suspended_job_ded_time_calendared(self):
+        """
+        This test confirms that a suspended job becomes top job when unable to
+        resume due to conflict with dedicated time.
+        """
+        now = time.time()
+        self.scheduler.add_dedicated_time(start=now + 180, end=now + 300)
+
+        j1 = Job(TEST_USER)
+        j1.set_attributes({ATTR_l + '.select': '1:ncpus=2',
+                           ATTR_l + '.walltime': 120})
+        jid1 = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        j2 = Job(TEST_USER)
+        j2.set_attributes({ATTR_l + '.select': '1:ncpus=1',
+                           ATTR_l + '.walltime': 60,
+                           ATTR_q: 'expressq'})
+        j2.set_sleep_time(60)
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid1)
+
+        starttime = time.time()
+        self.server.expect(JOB, {ATTR_state: 'F'}, id=jid2,
```

Comment:
I think you can just check the estimated_starttime of jid1 to be at or after "now +300" instead of waiting for the job to end.

Reply:
I removed the wait for the job to finish and now only checking if tracejob has "Job is a top job and will run at".